### PR TITLE
Add hCaptcha

### LINKED ISSUE #66: Add captcha to registration and contact form
Secure the following forms with a captcha:
- Registration
- Contact a person

Corresponding backend issue: https://github.com/icebreaker-science/backend/issues/60
Send the captcha toke in the body of the form

`{ "captcha" : token}
`

Use: Use https://www.hcaptcha.com/ for this.

### PR DESCRIPTION
**Pull request by dajenet**
Add captcha to registration and contact form

closes #66

### CODE REVIEW

File: src/app/contact-form/contact-form.component.ts [hunk 1]
```diff
@@ -32,21 +35,25 @@ export class ContactFormComponent implements OnInit {
...
+        (err) => {
+          this.emailError = err.error.message;
+          this.requestData.captcha = '';
+        },
       );
   }
+
+  // Captcha error
+  onError(error: any) {
+    this.emailError = 'Error while validating captcha.';
```
Comment (chaoran-chen) [inline src/app/contact-form/contact-form.component.ts]: 1. I think that this is a good time to rename the variable as it does not only store error messages concerning the email address.

2. Rename the function to onCaptchaError?

File: src/app/signup/signup.component.ts [hunk 2]
```diff
@@ -49,4 +51,9 @@ export class SignupComponent implements OnInit {
         },
       );
   }
+
+  // Captcha error
+  onError(error: any) {
```
Comment (chaoran-chen) [inline src/app/signup/signup.component.ts]: Same as above: rename to onCaptchaError?

**Review by chaoran-chen — APPROVED** [review]
Looks good! Only two small suggestions.

Please rebase and then feel free to merge :)

File: src/app/contact-form/contact-form.component.ts (hunk 1 above)
Comment (dajenet) [reply src/app/contact-form/contact-form.component.ts]: done

File: src/app/signup/signup.component.ts (hunk 2 above)
Comment (dajenet) [reply src/app/signup/signup.component.ts]: done

**Pull request merged**